purchases/src/test/java/com/revenuecat/purchases/PostingTransactionsTests.kt

```diff
@@ -372,12 +374,13 @@ class PostingTransactionsTests {
             onSuccess = { _, _ -> },
             onError = { _, _ -> }
         )
+
         verify(exactly = 1) {
             backendMock.postReceiptData(
                 purchaseToken = any(),
-                appUserID = appUserId,
+                appUserID = any(),
                 isRestore = any(),
-                observerMode = expectedConsumeAllTransactions,
+                observerMode = !expectedConsumeAllTransactions,
```

Comment:
I don't get what we were doing here 🤔 Were we sending `false` for observer mode?

Reply:
Oh this is a new test that is not in `main` so I assume it was wrong correct?

Reply:
yeah this is a new one that i somehow merged failing. looked back at the logic and fixed it here

Reply:
Hmm I think I would turn this around. I would rename to `expectedObserverMode`, set it to `false` and pass the negation to `postToBackend` since I think it's weird to negate the expectation, and in `postToBackend` it's not actually an expectation anyway.